doc/src/index-fileformats.md

```diff
@@ -0,0 +1,90 @@
+# Tantivy index file formats
+This document defines the index file formats used in Tantivy. 
+
+Tantivy has one metadata json file for each index and a few files for each segment, each file storing a specific datastructure.
+
+## Index metadata json
+For a given index, Tantivy stores the following metadata in a json file `meta.json` :
+- the list of segments with its metadata id, max_doc, deletes ;
+- the index schema represented by the list of fields with name, type and option data ;
+- the opstamp or timestamp of the operation (commit).
+
+```
+{
+  "segments": [
+    {
+      "segment_id": uuid string,
+      "max_doc": int|null,
+      "deletes": int|null
+    },
+    ...
+  ],
+  "schema": [
+    {
+      "name": string,
+      "type": string,
+      "options": {
+        "indexing": {
+          "record": string,
+          "tokenizer": string
+        },
+        "stored": bool
+      }
+    },
+ ],
+ ...
+ "opstamp": int
+}
+```
+
+
+## Summary of segment file extensions
+For a given segment, Tantivy stores a bunch of files whose name is set by segment uuid and whose extension defines the datastructure.
+
+| Name | Extension | Description |
+| --- | --- | --- |
+| Posting list | `.idx` | Sorted lists of document ids, associated to terms |
```

Comment:
I think we also need a separate chapter or section explaining what (Term, Posting etc...) are. Probably not in this PR.

Reply:
did you see the architecture file ? https://github.com/tantivy-search/tantivy/blob/main/ARCHITECTURE.md

Reply:
So my suggestion is that Architecture document could be moved here as part of this book. it gives more context on the vocabulary that is used in this chapter. Someone who has not read the architecture will not know what Term, Posting are unless he is familiar with search engine. I was also lost before reading this https://fulmicoton.com/posts/behold-tantivy/
maybe we can reference the architecture doc here?

Reply:
yes of course! I will add it.